static/js/http.js

```diff
@@ -14,13 +41,20 @@ function request (method, path, options = {}) {
   }
   return window.fetch(path, opts)
     .then(response => {
+      connectionStatus.lastSuccess = Date.now()
+      notifyConnectionStatusChange(true)
+
       if (!response.ok) {
         const error = { status: response.status, reason: response.statusText, is_error: true }
         return response.json()
           .then(json => { error.reason = json.reason; return error })
           .finally(() => { standardErrorHandler(error) })
       } else { return response.json().catch(() => null) }
     })
+    .catch(error => {
+      notifyConnectionStatusChange(false)
```

Comment:
Does this `catch` always mean server isn't avaialble or do we end up here on other errors as well like exceptions from the `then` case?